src/rotator_library/providers/iflow_provider.py

```
@@ -218,6 +234,58 @@ def _clean_schema_properties(self, properties: Dict[str, Any]) -> None:
+
+        # Handle reasoning_effort -> thinking conversion
+        reasoning_effort = payload.get("reasoning_effort")
+        if reasoning_effort:
+            budget = REASONING_BUDGETS.get(reasoning_effort, 8192)
+            payload["thinking"] = {"type": "enabled", "budget_tokens": budget}
+            payload.pop("reasoning_effort", None)
+            lib_logger.debug(f"Converted reasoning_effort={reasoning_effort} to thinking budget={budget}")
+        else:
+            # Default thinking for supported models (auto-enable)
+            payload["thinking"] = {"type": "enabled", "budget_tokens": -1}  # -1 = auto
+            lib_logger.debug(f"Auto-enabled thinking for model {model_id}")
```

Ruff (0.15.4)

[warning] Logging statement uses f-string

(G004)